core/alarms-lambda.js

```diff
@@ -23,7 +23,7 @@ module.exports = function LambdaAlarms (functionAlarmConfigs, context, serverles
     )
 
     for (const [funcResourceName, funcResource] of Object.entries(lambdaResources)) {
-      const functionName = funcResource.Properties.FunctionName
+      const functionName = funcResource.Properties.FunctionName ? funcResource.Properties.FunctionName : { Ref: funcResourceName }
```

Comment:
I think we should be able to rewrite this as:

```suggestion
      const functionName = funcResource.Properties.FunctionName || { Ref: funcResourceName }
```

if you also consider that as more readable

Reply:
done, but still having issues to get the generated alarm name correctly. it appears as '[LambdaDuration_[object Object]]'.
Not sure how I resolve that.

Reply:
See the `Sub` suggestion in the other comment

Reply:
corrected.